terraform/modules/aws-lambda-with-cron-schedule/20-lambda.tf

```diff
@@ -0,0 +1,40 @@
+locals {
+  lambda_name_underscore = replace(lower(var.function_name), "/[^a-zA-Z0-9]+/", "_")
+}
+
+data "archive_file" "lambda" {
+  type        = "zip"
+  source_dir  = "../../lambdas/${local.lambda_name_underscore}"
+  output_path = "../../lambdas/${local.lambda_name_underscore}.zip"
+}
+
+resource "aws_s3_bucket_object" "lambda" {
+  bucket = var.lambda_artefact_storage_bucket
+  key    = "${local.lambda_name_underscore}.zip"
+  source = data.archive_file.lambda.output_path
+  etag   = filemd5(data.archive_file.lambda.output_path)
+  acl    = "private"
+}
+
+resource "aws_lambda_function" "lambda" {
+  function_name     = lower("${var.identifier_prefix}${var.function_name}")
+  role              = aws_iam_role.lambda.arn
+  handler           = "main.lambda_handler"
+  runtime           = "python3.8"
```

Comment:
For flexibility would it be better to set this from input variable and support 3.9 as well if someone wants to use that? No point supporting 3.7 anymore I don't think.